code/modules/games/cards.dm

```diff
@@ -0,0 +1,435 @@
+/datum/playingcard
+	var/name = "playing card"
+	var/card_icon = "card_back"
+	var/back_icon = "card_back"
+
+/obj/item/weapon/deck
+	w_class = 2
+	icon = 'icons/obj/playing_cards.dmi'
+	var/list/cards = list()
+	var/cooldown = 0 // to prevent spam shuffle
+
+/obj/item/weapon/deck/holder
+	name = "card box"
+	desc = "A small leather case to show how classy you are compared to everyone else."
+	icon_state = "card_holder"
+
+/obj/item/weapon/deck/cards
+	name = "deck of cards"
+	desc = "A simple deck of playing cards."
+	icon_state = "deck"
+
+/obj/item/weapon/deck/cards/New()
+	..()
+	var/datum/playingcard/P
+	for(var/suit in list("spades","clubs","diamonds","hearts"))
+
+		var/colour
+		if(suit == "spades" || suit == "clubs")
+			colour = "black_"
+		else
+			colour = "red_"
+
+		for(var/number in list("ace","two","three","four","five","six","seven","eight","nine","ten"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]num"
```

Comment:
Won't this make all the icons the same regardless of the card? `black_num` or `red_num` or is that intentional?

Reply:
Certain it is intentional as there's no varied icons unlike tg card. However tg card once in a hand have a "Fake" sprite that doesn't show its contents, so to me it is an acceptable sacrifice.